src/main/kotlin/com/github/burrunan/s3cache/internal/AwsS3BuildCacheService.kt

```diff
@@ -229,26 +239,32 @@ class AwsS3BuildCacheService internal constructor(
                 gradleVersion = GradleVersion.current().toString()
             )
         }
-        metadata?.appendTo(meta.userMetadata)
+        val userMetadata = metadata?.appendTo(mutableMapOf())
```

Comment:
Maybe not for this PR but might change `CacheEntryMetadata` to return a map (`toMap()`) instead of `appendTo`

Reply:
Indeed. `toMap` is way easier to understand.